lib/traject/macros/iiif.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module Macros
+  # Macros for extracting Stanford Specific MODS values from Nokogiri documents
+  module IIIF
+    def grab_sul_iiif_links(manifest)
+      resp = Faraday.get URI.parse(manifest)
+      ::JSON.parse(resp.body)
+    rescue Faraday::Error
+      nil
+    end
+
+    def process_iiif_thumbnail(manifest_json)
+      manifest_json['thumbnail']['@id']
+    rescue ::JSON::ParserError
```

Comment:
Got it. Here are some ideas:

```
manifest_json.dig('thumbnail', '@id')
```
(dig is new-ish, I've never used it, but this seems like a great spot for it, if it works)

The old style:

```
manifest_json.fetch('thumbnail', {}).fetch('@id')
```

or with lots of guard clauses:

```
manifest_json['thumbnail']['@id'] if manifest_json['thumbnail']
```